drivers/fuel_gauge/bq27z746/emul_bq27z746.c

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright (c) 2023, ithinx GmbH
+ * Copyright (c) 2023, Tonies GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Emulator for bq27z746 fuel gauge
+ */
+
+#define DT_DRV_COMPAT ti_bq27z746
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(EMUL_BQ27Z746);
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/emul.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/i2c_emul.h>
+#include <zephyr/sys/byteorder.h>
+
+#include "bq27z746.h"
+
+/** Static configuration for the emulator */
+struct bq27z746_emul_cfg {
+	/** I2C address of emulator */
+	uint16_t addr;
+};
+
+static int emul_bq27z746_reg_write(const struct emul *target, int reg, int val)
+{
+	return -EIO;
+}
+
+static int emul_bq27z746_reg_read(const struct emul *target, int reg, int *val)
+{
+	switch (reg) {
+	case BQ27Z746_MANUFACTURERACCESS:
+		*val = 0;
```

Comment:
Totally fine with using stub values for now. I'm adding real values and behavior soon to the SBS emulator using a backend API.

However, could we change the stub value to something with bits that don't makeup a palindrome? That way we know the bytes are being sent back in the right order by the emulator. For the SBS emulator I just used `1`.

Reply:
Done. Also adapted the test so in case of emulation to test for the specific value, in case of real device test for ranges.